rfcs/01-http-hooks.md

```diff
@@ -0,0 +1,84 @@
+# HTTP hooks for integrating morph deployments into existing infrastructure
+
+## Summary
+
+We need a way of triggering external systems during deployments. We also want this to be generic, and promote a healthy way of collaborating on deployments (=> do not introduce any local state).
+
+Examples:
+- Trigger draining a kubelet prior to deployment
+- Mark hosts in maintenance before deployment, and mark as ready after health checks are OK
+- Provide a way of skipping certain hosts for deployment
+- Provide locking to away multiple people deploying the same host (or group of hosts)
+- ...
+
+Hooks should be composable, so that one hook point can call out to multiple hooks.
+
+
+## Overview of implementation
+
+Hooks can be split into two types:
+1.  Multi-host hooks
+1.  Single-host hooks
+
+Multi-host hooks would be applicable when selecting hosts to work on, while single-host hooks are relevant when dealing with the actual deployment of a host.
+
+The suggested implementation of the hooks are exactly the same, only differing in the actual payload passed around:
+
+Serialize the payload as JSON, and send it to the configured endpoint. The endpoint then transforms it, and returns a payload in the same format. Non `200 OK` responses causes deployments to stop, unless the user opts into ignoring the result of endpoints using a command line option.
+
+### Why not execute commands instead?
+
+The most straightforward way of implementing hooks would be by executing a command and sending all required state as an argument to the command.
```

Comment:
Would a task runner as repo driver such as `just` do?

Reply:
maybe rather than implementing hooks, morph would expose events?
Benefit: is not really tied to a specific server implementation. Just consumers.

Reply:
There is even an ideal candidate: https://cloudevents.io/

Reply:
Other than that, I like the proposal! :+1: